lib/collection/variable-scope.js

```diff
@@ -316,6 +316,26 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
         this.values.clear();
     },
 
+    /**
+     * Replace all variable names with their values in the given string or object
+     *
+     * @param {string|object} obj A string or an object to replace variables in
+     * @returns {string|object} The string or object with variables (if any) substituted with their values
+     */
+    replaceIn: function (obj) {
+        var transformed,
+            result;
+
+        if (typeof obj !== 'object') {
```

Comment:
Instead of this, try:
- if `obj` is string, transform to an array
- if `obj` is not of type `object`, bail out
- finally, replace substitutions

This helps to avoid unnecessarily ops for undefined, null, etc. values.